src/Chapter2.hs

```diff
@@ -348,8 +351,10 @@ Implement a function that returns only the first half of a given list.
 >>> firstHalf "bca"
 "b"
 -}
--- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
+firstHalf :: [a] -> [a]
+firstHalf l = 
+    let a = length l `div` 2 - 1
+    in subList 0 a l
```

Comment:
Nice usage of `subList` here! Though, `subList` does more than required, and here it's enough to _take_ the first half of the list.